pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompactmerge/UpsertCompactMergeTaskGenerator.java

```diff
@@ -296,18 +307,24 @@ public static SegmentSelectionResult processValidDocIdsMetadata(Map<String, Stri
       // variables to maintain current group sum
       long currentValidDocsSum = 0;
       long currentTotalDocsSum = 0;
+      double currentOutputSegmentSizeInBytes = 0.0;
 
       for (SegmentMergerMetadata segment : segments) {
         long validDocs = segment.getValidDocIds();
         long invalidDocs = segment.getInvalidDocIds();
+        double expectedSegmentSizeInBytes =
+            (segment.getSegmentZKMetadata().getSizeThresholdToFlushSegment() * validDocs * 1.0)
```

Comment:
`getSizeThresholdToFlushSegment` returns the number of rows not the size in bytes.